atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/schema/KeyValueServiceValidator.java

```diff
@@ -114,31 +113,27 @@ private void validateTables(Set<TableReference> tables) {
 
     private void validateTable(final TableReference table) {
         final int limit = getBatchSize(table);
-        // read only, but need to use a write tx in case the source table has SweepStrategy.THOROUGH
-        validationFromTransactionManager.runTaskWithRetry(
-                (TransactionTask<Map<Cell, byte[]>, RuntimeException>) t1 -> {
-                    validateTable(table, limit, t1);
-                    return null;
-                });
+        byte[] nextRowName = new byte[0];
+        while (nextRowName != null) {
+            nextRowName = validateNextRow(table, limit, nextRowName);
+        }
         KeyValueServiceMigratorUtils
                 .processMessage(messageProcessor, "Validated " + table, KvsMigrationMessageLevel.INFO);
     }
 
-    private void validateTable(final TableReference table, final int limit, final Transaction t1) {
-        // read only, but need to use a write tx in case the source table has SweepStrategy.THOROUGH
-        validationToTransactionManager.runTaskWithRetry(
-                (TransactionTask<Map<Cell, byte[]>, RuntimeException>) t2 -> {
-                    validateTable(table, limit, t1, t2);
-                    return null;
-                });
-    }
-
-    private void validateTable(TableReference table, int limit, Transaction t1, Transaction t2) {
-        RangeRequest.Builder builder = RangeRequest.builder().batchHint(limit);
-        byte[] nextRowName = new byte[0];
-        while (nextRowName != null) {
-            RangeRequest range = builder.startRowInclusive(nextRowName).build();
-            nextRowName = validateAndGetNextRowName(table, limit, t1, t2, range);
+    private byte[] validateNextRow(TableReference table, int limit, byte[] nextRowName) {
+        try {
+            return validationFromTransactionManager.runTaskWithRetry(t1 ->
```

Comment:
This change means that with default batch size, we'll do a transaction each 100 rows, so it will be probably quite slow.

Reply:
This should be fine, most people override this anyway. And it's run with embedded timelock usually, so overhead should be minimal.